kernel/src/main/scala/cats/Eq.scala

```diff
@@ -0,0 +1,106 @@
+package cats
+
+import scala.{specialized => sp}
+
+import scala.math.Equiv
+
+/**
+ * A type class used to determine equality between 2 instances of the same
+ * type. Any 2 instances `x` and `y` are equal if `eqv(x, y)` is `true`.
+ * Moreover, `eqv` should form an equivalence relation.
+ */
+trait Eq[@sp A] extends Any with Serializable { self =>
```

Comment:
> Does Simulacrum work with specialization, and could we use it here?

Yes it does. Also, simulacrum 0.6.0-SNAPSHOT makes type classes extend `Any with Serializable`.